test/test_datasets.py

```diff
@@ -1755,5 +1755,43 @@ def test_images_download_preexisting(self):
                 pass
 
 
+class INaturalistTestCase(datasets_utils.ImageDatasetTestCase):
+    DATASET_CLASS = datasets.INaturalist
+    FEATURE_TYPES = (PIL.Image.Image, (int, tuple))
+
+    ADDITIONAL_CONFIGS = datasets_utils.combinations_grid(
+        target_type=("kingdom", "full", "genus", ["kingdom", "phylum", "class", "order", "family", "genus", "full"]),
+        version=("2021_train",),
```

Comment:
For the future, it would be good to also test for the other years, as they contain different code-paths in the initialization phase